jsonparse.js

```diff
@@ -261,17 +266,19 @@ proto.write = function (buffer) {
             break;
           default:
             this.tState = START;
-            var result = Number(this.string);
-
-            if (isNaN(result)){
+            if (isNaN(this.string)){
```

Comment:
[isNaN will first coerce non-numeric values into numbers](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/isNaN#Confusing_special-case_behavior). This means that this check will pass if we get and empty string or a boolean.

This behavior should match what was already happening.